### PR TITLE
Moving hotspot rules to dedicated folders

### PR DESCRIPTION
**Pull request by csaba-sagi-sonarsource**

### CODE REVIEW

**Comment by Corniel** [comment]
Just out of curiosity, what do you want to achieve with this? If a rule is a _hotspot_ or not feels quite arbitrary.

File: analyzers/tests/SonarAnalyzer.UnitTest/Rules/Hotspots/NonStandardCryptographicAlgorithmsShouldNotBeUsedTest.cs [hunk 1]
```diff
@@ -33,28 +33,28 @@ public class NonStandardCryptographicAlgorithmsShouldNotBeUsedTest
         [TestMethod]
         [TestCategory("Rule")]
         public void NonStandardCryptographicAlgorithmsShouldNotBeUsed_CSharp() =>
-            Verifier.VerifyAnalyzer(@"TestCases\NonStandardCryptographicAlgorithmsShouldNotBeUsed.cs",
+            Verifier.VerifyAnalyzer(@"TestCases\Hotspots\NonStandardCryptographicAlgorithmsShouldNotBeUsed.cs",
```
Comment (pavel-mikula-sonarsource) [inline analyzers/tests/SonarAnalyzer.UnitTest/Rules/Hotspots/NonStandardCryptographicAlgorithmsShouldNotBeUsedTest.cs:36]: While looking at the files in the directory, this rule has too long name. Let's rename this type and all related stuff to `UsingNonstandardCryptography`

**Review by pavel-mikula-sonarsource — APPROVED** [review]
LGTM. Let's rename one of the rules while we're on move/rename topic.

Master if frozen now, so let's don't merge it before version bumping.

**Comment by csaba-sagi-sonarsource** [comment]
> Just out of curiosity, what do you want to achieve with this? If a rule is a _hotspot_ or not feels quite arbitrary.
Our plan is to organize a bit better the solution, and move things into subfolders and that way increase the navigability.

**Pull request merged**